tic_tac_toe/game.py

```diff
@@ -1,104 +1,159 @@
+    for i in range(0, len(board)):
+        first_diagonal.append((i,i))
+    test_first_diagonal = tuple(first_diagonal)    
+    if _is_winning_combination(board, test_first_diagonal, player):
+        return player
+
+    #for second diagonal
+    for i in range(0, len(board)):
+        second_diagonal.append((i,(len(board) - 1) - i))
+    test_second_diagonal = tuple(second_diagonal)
+    if _is_winning_combination(board, test_second_diagonal, player):
+        return player
+
+    return None
+    
 # public interface
 def start_new_game(player1, player2):
-    """
-    Creates and returns a new game configuration.
-    """
-    pass
-
+    return {'player1' : "X", 'player2' : "O", 'board' : [["-","-","-"],["-","-","-"],["-","-","-"]]
+     , 'next_turn' : "X", 'winner' : None}
+    
 
 def get_winner(game):
-    """
-    Returns the winner player if any, or None otherwise.
-    """
-    pass
+    return game['winner']
 
 
 def move(game, player, position):
-    """
-    Performs a player movement in the game. Must ensure all the pre requisites
-    checks before the actual movement is done.
-    After registering the movement it must check if the game is over.
-    """
-    pass
-
+   
+    #errors:
+    #game over exception
+    if _board_is_full(game['board']):
+        raise InvalidMovement('Game is over.')
+    #board not full but someone else won
+    if get_winner(game) is not None:
+        raise InvalidMovement('Game is over.')
+    #wrong turn error
+    if game['next_turn'] != player:
+        raise InvalidMovement('"%s" moves next.' % game['next_turn'])
+    
+    #position already taken error
+    if not _position_is_empty_in_board(position, game['board']):
+        raise InvalidMovement('Position already taken.')
+    
+    #position out of range error
+    if not _position_is_valid(position):
+        raise InvalidMovement('Position out of range')
+        
+    #actual move function
+    game['board'][position[0]][position[1]] = player
+    
+    if player=="X":
+        game['next_turn'] = "O"
+    else:
+        game['next_turn'] = "X"
+    
+    #check who winner is
+    #if board is full:
+    if (_check_winning_combinations(game['board'], player)):
+        game['winner'] = player
+        raise GameOver('"%s" wins!' % game['winner'])
+    elif _board_is_full(game['board']):
+        raise GameOver('Game is tied!')
+            
 
 def get_board_as_string(game):
-    """
-    Returns a string representation of the game board in the current state.
-    """
-    pass
-
+    pipe = '  |  '
+    dashes = '-'*14
+    #new_list = [y for x in game['board'] for y in x]
+    return '\n{}{}{}{}{}\n{}\n{}{}{}{}{}\n{}\n{}{}{}{}{}\n'.format(
+        game['board'][0][0], pipe, game['board'][0][1], pipe, game['board'][0][2], 
+        dashes, 
+        game['board'][1][0], pipe, game['board'][1][1], pipe, game['board'][1][2], 
+        dashes, 
+        game['board'][2][0], pipe, game['board'][2][1], pipe, game['board'][2][2])
 
```

Comment:
Wow, thank you very much. This was our "every other test passed and we're so tired, lets just put this in" last ditch effort lol. 
Looking at everyone else's code all I see are different ways I would have rather written certain sections. Thanks for the comment!